README.md

```diff
@@ -26,4 +26,66 @@ To test in a local Jenkins instance
 
     mvn hpi:run
 
-  [wiki]: http://wiki.jenkins-ci.org/display/JENKINS/SCM+API+Plugin
\ No newline at end of file
+  [wiki]: http://wiki.jenkins-ci.org/display/JENKINS/SCM+API+Plugin
+
+# SCM API Data Model
+
+## SCMSource
+
+[SCMSource](src/main/java/jenkins/scm/api/SCMSource.java)
+[(doc)](http://javadoc.jenkins.io/plugin/scm-api/jenkins/scm/api/SCMSource.html)
+implementations provide a configuration form (as a
+[Descriptor](http://javadoc.jenkins-ci.org/hudson/model/Descriptor.html))
+for users, and define the methods for multi-branch and other advanced
+SCM concepts in Jenkins.
+
+The most crucial methods an
+[SCMSource](src/main/java/jenkins/scm/api/SCMSource.java) must
+implement are:
+
+* [retrieve(HeadObserver, TaskListener)](https://github.com/jenkinsci/scm-api-plugin/blob/c06684f6e252bd4f8ef99ab8ec6eefceed8dce45/src/main/java/jenkins/scm/api/SCMSource.java#L153) - provides HeadObserver with the Set(name, hash) of revisions to consider
+* [retrieve(SCMHead, TaskListener)](https://github.com/MarkEWaite/scm-api-plugin/blob/c06684f6e252bd4f8ef99ab8ec6eefceed8dce45/src/main/java/jenkins/scm/api/SCMSource.java#L222) - returns a single revision
+* [retrieve(String, TaskListener)](https://github.com/MarkEWaite/scm-api-plugin/blob/c06684f6e252bd4f8ef99ab8ec6eefceed8dce45/src/main/java/jenkins/scm/api/SCMSource.java#L282) - returns a single revision identified by the string value
+
+## SCMHead
+
+[SCMHead](src/main/java/jenkins/scm/api/SCMHead.java)
```

Comment:
Also need to note that the `SCMHead` is treated like a key, so the equals contract is special. It should be treated as a "string" with type info rather than a store of data. The mixin interfaces can put additional constraints on equality